ethcore/src/json_tests/test_common.rs

```diff
@@ -30,43 +30,60 @@ pub enum HookType {
 	OnStop
 }
 
+/// Run all tests under the given path (except for the test files named in the skip list) using the
+/// provided runner function.
 pub fn run_test_path<H: FnMut(&str, HookType)>(
-	p: &Path, skip: &[&'static str],
-	runner: fn(json_data: &[u8], start_stop_hook: &mut H) -> Vec<String>,
+	path: &Path,
+	skip: &[&'static str],
+	runner: fn(path: &Path, json_data: &[u8], start_stop_hook: &mut H) -> Vec<String>,
 	start_stop_hook: &mut H
 ) {
+	if !skip.is_empty() {
+		// todo[dvdplm] it's really annoying to have to use flushln here. Should be `info!(target: "json-tests", …)`
+		flushln!("[run_test_path] Skipping tests in {}: {:?}", path.display(), skip);
```

Comment:
why is `flushln!` needed? Isn't `info! output unbuffered` (i.e, wait for the entire output before continuing)?

Reply:
The json test logging needs a major overhaul and I did not want to spend too much time on that at this time. `println!` was swallowed, as was `info!`. I think it has to do with the setup where a string is `flush!` at the start of a test; at the end it is completed with an `ok` or `failed`. It really should be all `info!`s, `warn!` etc. I tried to port them all over but it did not work right away so I reverted and kept the weirdness without digging deeper.

Reply:
Ok, let's keep it out of this PR and create a separate issue? Nice, that you have investigated it though..